feat: UTC-416: Implement 'Ensure all annotators are evaluated' setting for GT tasks

mcanu opened the pull request:
<!-- Follow Merge -->
This PR was automatically generated via Follow Merge. [Workflow Run](https://github.com/HumanSignal/label-studio/actions/runs/20244404085)
Please ensure that all linked upstream Pull Requests are merged before proceeding with this one.
- https://github.com/HumanSignal/label-studio-sdk/pull/690
<!-- Follow Merge -->

mcanu committed:
Added show_ground_truth_always field in project and handle next_task

mcanu committed:
Modified migration

mcanu committed:
Fix migration default

mcanu committed:
Fixed migration

nass600 commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/19924849173)

nass600 committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/19924849173

nass600 committed:
Merge branch 'develop' into 'fb-utc-416'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/19924849173

nass600 committed:
added warning badge

nass600 committed:
added icons

mcanu committed:
Added new fields

matt-bernstein commented:
/git merge develop
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/19972691403)
> Successfully merged:  10 files changed, 161 insertions(+), 69 deletions(-)

matt-bernstein committed:
Merge branch 'develop' into 'fb-utc-416'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/19972691403

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/19972821374

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/19972907652

mcanu committed:
Using new fields

matt-bernstein commented on label_studio/projects/api.py:
```diff
@@ -308,6 +308,7 @@ def get_queryset(self):
                             'total_predictions_number': 0,
                             'sampling': 'Sequential sampling',
                             'show_ground_truth_first': True,
+                            'annotator_evaluation_enabled': False,
```
does this change the default for new projects?

matt-bernstein commented on label_studio/projects/functions/next_task.py:
```diff
@@ -81,10 +78,10 @@ def _try_tasks_with_overlap(tasks: QuerySet[Task]) -> Tuple[Union[Task, None], Q
+    # Exclude ground truth annotations from the count when not in annotator evaluation mode
     # to prevent GT tasks from being prioritized via breadth-first logic
     annotation_filter = ~Q(annotations__completed_by=user)
-    if not project.show_ground_truth_first:
```
is this setting used for other things or is it deprecated now? If deprecated, lave a comment somewhere visible

matt-bernstein replied:
In the API docs for this parameter, eg

matt-bernstein commented on label_studio/projects/models.py:
```diff
@@ -313,11 +313,21 @@ class SkipQueue(models.TextChoices):
         max_length=100, choices=SkipQueue.choices, null=True, default=SkipQueue.REQUEUE_FOR_OTHERS
     )
+
+    # Deprecated
```
ah, I see :) wasn't there some django convention that you introduced for deprecated features, or am I misremembering?

matt-bernstein replied:
Ah, here it is
```
@extend_schema_serializer(deprecate_fields=['agreement_threshold', 'max_additional_annotators_assignable'])
class MetricParamSerializer(serializers.ModelSerializer):
    ...
```

mcanu replied:
<img width="931" height="124" alt="image" src="https://github.com/user-attachments/assets/ec139c3c-e2dd-40a0-a113-280e75505058" />
Its marked as deprecated in the serializer

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/19974215346

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/19974288596

mcanu commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/20031202667)

mcanu committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/20031202667

mcanu committed:
Merge branch 'develop' into 'fb-utc-416'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/20031202667

mcanu committed:
Deprecated show_ground_truth_first

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/20044558200

robot-ci-heartex committed:
Merge branch 'develop' into 'fb-utc-416'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/20044558200

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/20044652944

mcanu committed:
Fix test

mcanu commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/20045038930)

mcanu committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/20045038930

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/20045264167

mcanu commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/20048026880)

mcanu committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/20048026880

mcanu commented:
/git merge develop
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/20071148448)
> Successfully merged:  create mode 100644 label_studio/io_storages/localfiles/README.md

mcanu committed:
Merge branch 'develop' into 'fb-utc-416'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/20071148448

mcanu committed:
annotator_evaluation_enabled default to False

mcanu committed:
Fix columns test

mcanu commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/20100599882)

mcanu committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/20100599882

mcanu committed:
Merge branch 'develop' into 'fb-utc-416'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/20100599882

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/20100847806

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/20100945769

bmartel commented on label_studio/projects/functions/next_task.py:
```diff
@@ -283,7 +289,7 @@ def get_next_task_without_dm_queue(
         # if there are already labeled tasks, but task.overlap still < project.maximum_annotations, randomly sampling from them
         logger.debug(f'User={user} tries depth first from prepared tasks')
-        next_task = _try_breadth_first(not_solved_tasks, user, project)
+        next_task = _try_breadth_first(not_solved_tasks, user, project, attempt_gt_first)
```
```
        next_task = _try_breadth_first(not_solved_tasks, user, project, attempt_gt_first)
```

This passing of attempt_gt_first to the other functions are the secret sauce it would seem. Nice solution 🙌

bmartel reviewed (approved):
Looks like a very precise change, well thought out!

mcanu commented:
/git merge develop
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/20144673543)
> Successfully merged:  create mode 100644 web/libs/ui/src/assets/icons/state-needs-review.svg

mcanu committed:
Merge branch 'develop' into 'fb-utc-416'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/20144673543

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/20145073906

robot-ci-heartex committed:
Merge branch 'develop' into 'fb-utc-416'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/20145073906

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/20145278941

mcanu commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/20180337188)

mcanu committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/20180337188

mcanu committed:
Merge branch 'develop' into 'fb-utc-416'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/20180337188

mcanu commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/20234412494)

mcanu committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/20234412494

mcanu committed:
Merge branch 'develop' into 'fb-utc-416'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/20234412494

mcanu commented:
/git merge develop
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/20244254420)
> Successfully merged:  12 files changed, 200 insertions(+), 21 deletions(-)

mcanu committed:
Merge branch 'develop' into 'fb-utc-416'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/20244254420

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/20244404085

mcanu commented:
/git merge develop
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/20249047965)
> Successfully merged:  3 files changed, 49 insertions(+), 5 deletions(-)

mcanu committed:
Merge branch 'develop' into 'fb-utc-416'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/20249047965

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio-enterprise/actions/runs/20270438305
